distribution/tools/entitlement-agent/src/main/java/org/elasticsearch/entitlement/agent/ConfigurationScanner.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.agent;
+
+import org.elasticsearch.entitlement.checks.CheckBefore;
+import org.elasticsearch.entitlement.instrumentation.MethodKey;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Collections.unmodifiableSet;
+
+/**
+ * Uses reflection to scan for instrumentation method definitions,
+ * which establish which methods we should annotate and how.
+ */
+public class ConfigurationScanner {
```

Comment:
I find this class not necessary. We will very likely end up using something different (more human/machine friendly) to represent the list of methods we need to instrument, so for now it's better to stay simple.
I would remove ConfigurationScanner, CheckBefore and EntitlementChecks, and just build by hand a "ScanResult" (which will contain the single entry for System.exit), and keep the PR a bit simpler and focused.

Reply:
I had this before, but I moved to this system to better reflect what we decided at the last sync meeting.